### PR TITLE
Move colors to css vars

### PR DESCRIPTION
**Pull request by thislooksfun**
This is the first step towards [respecting the OS's theme](https://github.com/TypeStrong/typedoc/projects/6#card-51842790)

NOTE: This is just a first draft. There are still some hard-coded colors (mainly in shadows) that should be moved out to variables as well. You can find them easily with the regex `:.*(rgba?\(|#(?!\{))`. I didn't split those into vars because I wasn't sure exactly what to call them.

### CODE REVIEW

File: src/default/assets/css/_theme.scss [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Default light theme
+:root {
+    --td-color-background: #fdfdfd;
```
Comment (Gerrit0) [inline src/default/assets/css/_theme.scss]: I'd like to get rid of the `td` prefix. I've never liked the `tsd` prefix we use on all the classes, and once users can easily add custom CSS without creating a custom theme (https://github.com/TypeStrong/typedoc/issues/1060), I want to be able to maintain a pretty stable set of names.

**Review by Gerrit0 — CHANGES_REQUESTED** [review]

**Comment by Gerrit0** [comment]
Looking good! Moving the other colors can certainly wait for now.

File: src/default/assets/css/_theme.scss (hunk 1 above)
Comment (thislooksfun) [reply src/default/assets/css/_theme.scss]: Fair enough! I wasn't a big fan of them either, but I put them in to be in line with the `tsd` prefixes. I'll get rid of 'em.

Comment (thislooksfun) [reply src/default/assets/css/_theme.scss]: Resolved in c29d178dc85abc5db5840b55b5990fed9d7a639c!

**Review by Gerrit0 — APPROVED** [review]

**Pull request merged**